ospfd/ospf_lsa.c

```diff
@@ -587,14 +600,20 @@ static int lsa_link_broadcast_set(struct stream **s, struct ospf_interface *oi)
 	}
 	/* Describe type 3 link. */
 	else {
-		if (IS_DEBUG_OSPF(lsa, LSA_GENERATE))
-			zlog_debug(
-				"LSA[Type1]: Interface %s has no DR. Adding stub interface",
-				oi->ifp->name);
-		masklen2ip(oi->address->prefixlen, &mask);
-		id.s_addr = oi->address->u.prefix4.s_addr & mask.s_addr;
-		return link_info_set(s, id, mask, LSA_LINK_TYPE_STUB, 0,
-				     oi->output_cost);
+		if (OSPF_IF_PARAM(oi, prefix_suppression)) {
+			if (IS_DEBUG_OSPF(lsa, LSA_GENERATE))
+				zlog_debug("LSA[Type1]: Interface %s stub link omitted due prefix-suppression",
+					   oi->ifp->name);
+			return (0);
```

Comment:
nit: Usually `()` not needed to wrap.

Reply:
Changed in two places. The enclosing () were part of the coding style for the previous software base I worked on.